pkg/cli/validate.go

```diff
@@ -0,0 +1,194 @@
+`, "\n"),
+		Args: cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runValidateCmd(cmd, opts, args)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&opts.token, "token", "t", "", `
+The JVS token that needs validation, can be passed as string or via pipe
+`)
+	cmd.MarkFlagRequired("token") //nolint // not expect err
+	// TODO: #140 add output format flag.
+	return cmd
+}
+
+func runValidateCmd(cmd *cobra.Command, opts *validateCmdOptions, args []string) error {
+	ctx := context.Background()
+
+	jvsclient, err := client.NewJVSClient(ctx, &client.JVSConfig{
+		Version:         "1",
+		JWKSEndpoint:    opts.config.JWKSEndpoint,
+		CacheTimeout:    5 * time.Minute,
+		AllowBreakglass: true,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create jvs client: %w", err)
+	}
+
+	if opts.token == "-" {
+		buf, err := io.ReadAll(io.LimitReader(cmd.InOrStdin(), 64*1_000))
+		if err != nil || len(buf) == 0 {
+			fmt.Print("Enter token: ")
+			fmt.Scanf("%s", &opts.token)
+		} else {
+			opts.token = string(buf)
+		}
+	}
+
+	// Validate token
+	breakglass := false
+	token, err := jvspb.ParseBreakglassToken(opts.token)
+	if err != nil {
+		return fmt.Errorf("failed to parse breakglass token: %w", err)
+	}
+	if token != nil {
+		breakglass = true
+	} else {
+		token, err = jvsclient.ValidateJWT(opts.token)
+		if err != nil {
+			return fmt.Errorf("failed to validate jwt: %w", err)
+		}
+	}
+
+	// Output the token into three subtables: breakglass, justification,
+	// and standard claims.
+	w := tabwriter.NewWriter(cmd.OutOrStdout(), 0, 0, 2, ' ', 0)
+	if _, err := fmt.Fprintf(w, "-----BREAKGLASS-----\n%t\n", breakglass); err != nil {
+		return err
+	}
+
+	if err := writeJustification(w, token); err != nil {
+		return err
+	}
+	if err := writeStandardClaims(ctx, w, token); err != nil {
+		return err
+	}
+	w.Flush()
+	return nil
+}
+
+func writeJustification(w *tabwriter.Writer, token jwt.Token) error {
+	if _, err := fmt.Fprintln(w, "\n----JUSTIFICATION----"); err != nil {
+		return err
+	}
+	justs, err := jvspb.GetJustifications(token)
+	if err != nil {
+		return fmt.Errorf("failed to get justifications from token")
+	}
+	for _, j := range justs {
+		if _, err := fmt.Fprintf(w, "%s\t\"%s\"\n", j.GetCategory(), j.GetValue()); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeStandardClaims(ctx context.Context, w *tabwriter.Writer, token jwt.Token) error {
+	// Convert standard claims into a map, excluding justifications claim which is already handled
+	claimsMap, err := token.AsMap(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to convert token into map: %w", err)
+	}
+	delete(claimsMap, "justs")
+	claims := make(map[string]string, len(claimsMap))
+	claimsKeys := make([]string, 0, len(claimsMap))
+	for k, v := range claimsMap {
+		claimsKeys = append(claimsKeys, k)
+		jv, err := json.Marshal(v)
+		if err != nil {
+			claims[k] = fmt.Sprintf("failed to marshal to json: %s", err)
```

Comment:
Why don't we return the error?

Reply:
Yeah, we could, to be consistent, updating to return error.